module3/src/MyBinarySearchTree.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Created by yurii.pyvovarenko on 11/28/2021.
+ * MyBinarySearchTree without balancing
+ */
+public class MyBinarySearchTree<T extends Comparable<T>> {
+    private static final String SPACE = " ";
+    private Node<T> root;
+
+    public MyBinarySearchTree(T value) {
+        root = new Node<T>(value);
+    }
+
+    void insert(T value)  {
+        root = insertRecursively(root, value);
+    }
+
+    Node insertRecursively(Node<T> root, T value) {
+        //tree is empty
+        if (root == null) {
+            root = new Node(value);
+            return root;
+        }
+        //traverse the tree
+        if (value.compareTo(root.getValue()) < 0) { //insert in the left subtree
+            root.setLeft(insertRecursively(root.getLeft(), value));
+            System.out.println("insert left. value = " + value + ", root.getValue() = " + root.getValue());
+        } else if (value.compareTo(root.getValue()) > 0) {    //insert in the right subtree
+            root.setRight(insertRecursively(root.getRight(), value));
+            System.out.println("insert right. value = " + value + ", root.getValue() = " + root.getValue());
```

Comment:
please use `logback` instead of `sop` in the next modules. You need to created `maven` or `gradle` project for that purpose

Reply:
Ok. I'll try it in the next project.

Reply:
https://logback.qos.ch/setup.html

Reply:
use maven build - https://logback.qos.ch/setup.html#mavenBuild